lib/govuk_publishing_components/presenters/step_by_step_nav_helper.rb

```diff
@@ -57,7 +57,7 @@ def list(element)
       def create_list_item_content(link)
         if link[:href]
           @link_index += 1
-          href = link_href(link[:active], link[:href])
+          href = link_href(link[:active], link[:href], "step-by-step-nav" => @options[:step_nav_content_id])
```

Comment:
I don't think you need to add pass the query parameters in here as the step nav id is always going to added to the URL. You can do it directly in `update_href_with_parameters`.